pkg/authorization/toplevel_org_authorizer.go

```diff
@@ -71,11 +77,21 @@ type topLevelOrgAccessAuthorizer struct {
 func (a *topLevelOrgAccessAuthorizer) Authorize(ctx context.Context, attr authorizer.Attributes) (authorized authorizer.Decision, reason string, err error) {
 	cluster, err := genericapirequest.ValidClusterFrom(ctx)
 	if err != nil || cluster == nil || cluster.Name.Empty() {
+		kaudit.AddAuditAnnotations(
+			ctx,
+			TopLevelContentAuditDecision, DecisionNoOpinion,
+			TopLevelContentAuditReason, fmt.Sprintf("error getting cluster from request: %v", err),
+		)
 		return authorizer.DecisionNoOpinion, WorkspaceAcccessNotPermittedReason, err
 	}
 
 	if !cluster.Name.HasPrefix(tenancyv1alpha1.RootCluster) {
 		// nobody other than system:masters (excluded from authz) has access to workspaces not based in root
+		kaudit.AddAuditAnnotations(
+			ctx,
+			TopLevelContentAuditDecision, DecisionNoOpinion,
+			TopLevelContentAuditReason, "only system:masters may access workspaces with a non-root prefix",
```

Comment:
this text is a resolution not a reason. The reason should rather be `toplevel.authorization.kcp.io/decision: non-root prefixed workspace access not permitted`.